File: app/src/main/java/com/example/juan/buscachinos/Controller.java
package com.example.juan.buscachinos;

import android.database.Cursor;
import android.util.SparseArray;

import java.util.ArrayList;

/**
 * Created by Juan on 22/06/2017.
 */

public class Controller {

    public int getCantidadCategorias() {
        int cont = 0;
        if (Constants.database != null) {
            Cursor c = Constants.database.rawQuery("SELECT * FROM chino", null);

            cont = c.getCount();
        }
        return cont;
    }

    public ArrayList<Chino> getChinos(){
        ArrayList<Chino> chinos = new ArrayList<Chino>();
        Chino chino = null;
        if (Constants.database != null) {
            Cursor c = Constants.database.rawQuery("SELECT * FROM chino ", null);
            int i = 0;
            if (c.moveToFirst()) {
                do {
                    chino = new Chino();

                    final int codChino = Integer.parseInt(c.getString(0));
                    final String chino_name = c.getString(1);
                    final double longitud = Double.parseDouble(c.getString(2));
                    final double latitude = Double.parseDouble(c.getString(3));

                    chino.setCodChino(codChino);
                    chino.setChino_name(chino_name);
                    chino.setLongitud(longitud);
                    chino.setLatitude(latitude);
                    chinos.add(chino);
                    i++;
                } while (c.moveToNext());
            }
        }
        return chinos;
    }

    public Chino getChinobyCoords(double longitud, double latitud){
        Chino chino = null;
        if (Constants.database != null) {
            final Cursor c = Constants.database.rawQuery("SELECT * FROM chino WHERE longitud = "+longitud+" AND latitud = " + latitud, null);
            //txtResultado.setText("");


            int i = 0;
            if (c.moveToFirst()) {
                do {
                    chino = new Chino();
                    chino.setCodChino(Integer.parseInt(c.getString(0)));
                    chino.setChino_name(c.getString(1));
                    chino.setLongitud(Double.parseDouble(c.getString(2)));
                    chino.setLatitude(Double.parseDouble(c.getString(3)));

                    i++;
                } while (c.moveToNext());
            }
        }
        return chino;
    }

    public Chino getChinoCod(int cod){
        Chino chino = new Chino();
        if (Constants.database != null) {
            final Cursor c = Constants.database.rawQuery("SELECT * FROM chino WHERE codChino = "+cod, null);
            //txtResultado.setText("");


            int i = 0;
            if (c.moveToFirst()) {
                do {
                    chino.setCodChino(Integer.parseInt(c.getString(0)));
                    chino.setChino_name(c.getString(1));
                    chino.setLongitud(Double.parseDouble(c.getString(2)));
                    chino.setLatitude(Double.parseDouble(c.getString(3)));

                    i++;
                } while (c.moveToNext());
            }
        }
        return chino;
    }

}

File: app/src/main/java/com/example/juan/buscachinos/Controller.kt
package com.example.juan.buscachinos

/**
 * Created by Juan on 22/06/2017.
 */
class Controller {

    fun getCantidadCategorias(): Int {
        val db = Constants.database ?: return 0
        db.rawQuery("SELECT * FROM chino", null).use { c ->
            return c.count
        }
    }

    fun getChinos(): ArrayList<Chino> {
        val chinos = ArrayList<Chino>()
        val db = Constants.database ?: return chinos
        db.rawQuery("SELECT * FROM chino", null).use { c ->
            if (c.moveToFirst()) {
                do {
                    chinos.add(
                        Chino(
                            codChino = c.getString(0).toInt(),
                            chino_name = c.getString(1),
                            longitud = c.getString(2).toDouble(),
                            latitude = c.getString(3).toDouble()
                        )
                    )
                } while (c.moveToNext())
            }
        }
        return chinos
    }

    fun getChinobyCoords(longitud: Double, latitud: Double): Chino? {
        val db = Constants.database ?: return null
        var chino: Chino? = null
        db.rawQuery(
            "SELECT * FROM chino WHERE longitud = $longitud AND latitud = $latitud",
            null
        ).use { c ->
            if (c.moveToFirst()) {
                do {
                    chino = Chino(
                        codChino = c.getString(0).toInt(),
                        chino_name = c.getString(1),
                        longitud = c.getString(2).toDouble(),
                        latitude = c.getString(3).toDouble()
                    )
                } while (c.moveToNext())
            }
        }
        return chino
    }

    fun getChinoCod(cod: Int): Chino {
        val chino = Chino()
        val db = Constants.database ?: return chino
        db.rawQuery("SELECT * FROM chino WHERE codChino = $cod", null).use { c ->
            if (c.moveToFirst()) {
                do {
                    chino.codChino = c.getString(0).toInt()
                    chino.chino_name = c.getString(1)
                    chino.longitud = c.getString(2).toDouble()
                    chino.latitude = c.getString(3).toDouble()
                } while (c.moveToNext())
            }
        }
        return chino
    }
}
